connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -504,6 +542,18 @@ class DataFrameReader private[sql] (sparkSession: SparkSession) extends Logging
     }
   }
 
+  private def parse(ds: Dataset[String], format: ParseFormat): DataFrame = {
+    sparkSession.newDataFrame { builder =>
+      val parseBuilder = builder.getParseBuilder
+        .setInput(ds.plan.getRoot)
+        .setFormat(format)
+      userSpecifiedSchema.foreach(schema => parseBuilder.setSchema(schema.toDDL))
```

Comment:
as to this PR itself, I think we should probably use `DataType schema` in the proto message, `schema.toDDL` always discards the metadata.

Reply:
What do you think about this? @hvanhovell

Reply:
[20f1722](https://github.com/apache/spark/pull/40332/commits/20f1722c4cc3331e0c8247f4d3cae419a3e81870) change to pass a DataType @zhengruifeng

Reply:
[ca6ec7b](https://github.com/apache/spark/pull/40332/commits/ca6ec7b4651b3e9c69bc96240687840179b49ff0) rename `data_type` to `schema` in proto message

Reply:
In this scenario, I think `DataType schema` is used and `schema.toDDL` is no longer needed